### PR TITLE
use k8s 6.0.0 and limit aws PV per node

### PR DESCRIPTION
**Pull request by calvix**
towards https://github.com/giantswarm/giantswarm/issues/8659

### CODE REVIEW

File: Gopkg.toml [hunk 1]
```diff
@@ -96,7 +96,7 @@ required = [
 
 [[override]]
   name = "github.com/giantswarm/k8scloudconfig"
-  version = "~0.1.0"
+  branch = "master"
```
Comment (tfussell) [inline Gopkg.toml]: I think this should still be `~0.1.0` until `aws-operator` switches to go modules. This probably works, but transitive dependencies might not be correct.

File: CHANGELOG.md [hunk 2]
```diff
@@ -8,6 +8,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ## [Unreleased]
 
 ## [8.2.1] - 2020-03-20
+- Add PV limit per node. The limit is 20 PV per node.
```
Comment (tfussell) [inline CHANGELOG.md]: This should go under `[Unreleased]` since 8.2.1 is already released I believe.

File: service/controller/internal/cloudconfig/tccp.go [hunk 3]
```diff
@@ -8,7 +8,8 @@ import (
 	infrastructurev1alpha2 "github.com/giantswarm/apiextensions/pkg/apis/infrastructure/v1alpha2"
 	g8sv1alpha1 "github.com/giantswarm/apiextensions/pkg/apis/provider/v1alpha1"
 	"github.com/giantswarm/certs"
-	k8scloudconfig "github.com/giantswarm/k8scloudconfig/v_5_2_0"
+	"github.com/giantswarm/k8scloudconfig/v_6_0_0"
+	k8scloudconfig "github.com/giantswarm/k8scloudconfig/v_6_0_0"
```
Comment (tfussell) [inline service/controller/internal/cloudconfig/tccp.go]: Double import?

File: service/controller/internal/cloudconfig/tccp_test.go [hunk 4]
```diff
@@ -11,7 +11,8 @@ import (
 
 	infrastructurev1alpha2 "github.com/giantswarm/apiextensions/pkg/apis/infrastructure/v1alpha2"
 	"github.com/giantswarm/certs"
-	k8scloudconfig "github.com/giantswarm/k8scloudconfig/v_5_2_0"
+	"github.com/giantswarm/k8scloudconfig/v_6_0_0"
+	k8scloudconfig "github.com/giantswarm/k8scloudconfig/v_6_0_0"
```
Comment (tfussell) [inline service/controller/internal/cloudconfig/tccp_test.go]: Imported twice.

File: service/controller/key/cluster.go [hunk 5]
```diff
@@ -14,7 +14,7 @@ import (
 const (
 	// CloudConfigVersion defines the version of k8scloudconfig in use. It is used
 	// in the main stack output and S3 object paths.
-	CloudConfigVersion = "v_5_2_0"
+	CloudConfigVersion = "v_6_0_0"
```
Comment (tfussell) [inline service/controller/key/cluster.go]: It might be cool to export this from `k8scloudconfig` in the future. One less thing to change.

File: service/controller/key/cluster.go [hunk 6]
```diff
@@ -29,6 +29,11 @@ const (
 	KubernetesSecurePort = 443
 )
 
+const (
+	KubectlVersion              = "9ccdc9dc55a01b1fde2aea73901d0a699909c9cd"
```
Comment (tfussell) [inline service/controller/key/cluster.go]: It would be nice to have a comment explaining which version this is just so we can check that it is compatible with the Kubernetes version.

File: service/controller/key/cluster.go (hunk 5 above)
Comment (MarcelMue) [reply service/controller/key/cluster.go]: Agreed - let's do that in a follow up PR though.

File: helm/aws-operator/templates/rbac.yaml [hunk 7]
```diff
@@ -5,6 +5,14 @@ metadata:
...
     {{- include "aws-operator.labels" . | nindent 4 }}
 rules:
+  - apiGroups:
+      - release.giantswarm.io
+    resources:
+      - release
+    verbs:
+      - get
+      - list
+      - watch
```
Comment (MarcelMue) [inline helm/aws-operator/templates/rbac.yaml]: We _should_ only need get - let's try this first though.

**Comment by MarcelMue** [comment]
Time to sort out go modules here :robot:

**Comment by MarcelMue** [comment]
@paurosello Do you have time to test this now?

**Comment by paurosello** [comment]
All good here! we can proceed

File: helm/aws-operator/templates/rbac.yaml (hunk 7 above)
Comment (paurosello) [reply helm/aws-operator/templates/rbac.yaml:15]: It works, do we want to restrict only to get?

Comment (MarcelMue) [reply helm/aws-operator/templates/rbac.yaml:15]: Yes, please. That should be enough.

**Comment by calvix** [comment]
@paurosello is this tested? did we solve both `k8s-network-setup` and also `wait-for-domains` issue?

**Comment by paurosello** [comment]
I booted a cluster and all good, did not know how to check for the wait-for-domains though

**Comment by calvix** [comment]
Can you guys explain to me where si thsi comming from?
```
github.com/giantswarm/k8scloudconfig/v6 v6.0.2-0.20200331075245-f7153f38c6d0
```

but in k8scloduconfig repo i only see tag 6.0.1 no branch or tag 6.0.2
Did i miss something?

File: helm/aws-operator/templates/rbac.yaml (hunk 7 above)
Comment (paurosello) [reply helm/aws-operator/templates/rbac.yaml]: Checking now

**Comment by MarcelMue** [comment]
> Can you guys explain to me where si thsi comming from?
> 
> ```
> github.com/giantswarm/k8scloudconfig/v6 v6.0.2-0.20200331075245-f7153f38c6d0
> ```
> 
> but in k8scloduconfig repo i only see tag 6.0.1 no branch or tag 6.0.2
> Did i miss something?

This was me to include your changes which were not tagged at the time in k8scc - they are only merged into the master branch.

See slack here:
https://gigantic.slack.com/archives/CSV04LUJK/p1585735546002700

Once your changes are tagged on k8scc we should be fine.

**Comment by MarcelMue** [comment]
What is stopping you from merging this? Can we help somehow?

File: go.mod [hunk 8]
```diff
@@ -20,7 +20,7 @@ require (
 	github.com/giantswarm/helmclient v0.2.0
 	github.com/giantswarm/ipam v0.2.0
 	github.com/giantswarm/k8sclient v0.2.0
-	github.com/giantswarm/k8scloudconfig/v6 v6.0.1
+	github.com/giantswarm/k8scloudconfig/v6 v6.0.2-0.20200331075245-f7153f38c6d0
```
Comment (tfussell) [inline go.mod:23]: Looks like this needs a release of v6.0.2 before merging.

File: pkg/project/project.go [hunk 9]
```diff
@@ -5,7 +5,7 @@ var (
 	gitSHA             = "n/a"
 	name        string = "aws-operator"
 	source      string = "https://github.com/giantswarm/aws-operator"
-	version            = "8.2.2-dev"
+	version            = "8.2.2"
```
Comment (tfussell) [inline pkg/project/project.go:8]: This should be changed in a separate commit. See https://gigantic.slack.com/archives/C04TGHDEF/p1583408026036700

Comment (paurosello) [reply pkg/project/project.go]: done

File: helm/aws-operator/templates/rbac.yaml (hunk 7 above)
Comment (paurosello) [reply helm/aws-operator/templates/rbac.yaml]: done

File: go.mod (hunk 8 above)
Comment (paurosello) [reply go.mod]: released and changed
